ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/activities/MapViewActivity.java

```diff
@@ -65,31 +78,105 @@ public Marker getLastMarker() {
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
-		// Map initialization taken from the osmdroid bonus pack tutorials
+		// Map initialization and event handling
+		// taken from the osmdroid bonus pack tutorials
 		// at https://code.google.com/p/osmbonuspack/wiki/Tutorial_0
+		// https://code.google.com/p/osmbonuspack/wiki/Tutorial_1
+		// and https://code.google.com/p/osmbonuspack/wiki/Tutorial_5
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_map_view);
 		
+		map = (MapView) findViewById(R.id.map);
+		map.setTileSource(TileSourceFactory.MAPNIK);
+		map.setBuiltInZoomControls(true);
+		map.setMultiTouchControls(true);
+		
 		Intent intent = getIntent();
 		Bundle bundle = intent.getExtras();
 		if (bundle != null) {
-			double latitude = intent.getDoubleExtra("latitude", Coordinate.DEFAULT_COORDINATE.getLatitude());
-			double longitude = intent.getDoubleExtra("longitude", Coordinate.DEFAULT_COORDINATE.getLongitude());
-			coordinate.setLatitude(latitude);
-			coordinate.setLongitude(longitude);
-			requestCode = intent.getIntExtra("requestCode", 0);
-			map = (MapView) findViewById(R.id.map);
-			map.setTileSource(TileSourceFactory.MAPNIK);
-			map.setBuiltInZoomControls(true);
-			map.setMultiTouchControls(true);
 
 			MapEventsOverlay mapEventsOverlay = new MapEventsOverlay(this, this);
 			map.getOverlays().add(0, mapEventsOverlay);
-			
-			GeoPoint startPoint = new GeoPoint(coordinate.getLatitude(), coordinate.getLongitude());
-			IMapController mapController = map.getController();
-			mapController.setZoom(18);
-			mapController.setCenter(startPoint);
+			requestCode = intent.getIntExtra("requestCode", 0);
+			if (requestCode == GeolocationRequestCode.SET_GEOLOCATION) {
+				double latitude = intent.getDoubleExtra("latitude", Coordinate.DEFAULT_COORDINATE.getLatitude());
```

Comment:
`"latitude"` and `"longitude"` should be put into the `IntentExtraIDs` class with the rest of 'em.